pkg/validatormanager/validatormanager.go

```diff
@@ -31,36 +42,238 @@ func AddPoAValidatorManagerContractToAllocations(
+	type InitialValidator struct {
+		NodeID       []byte
+		BlsPublicKey []byte
+		Weight       uint64
+	}
+	type SubnetConversionData struct {
+		SubnetID                     [32]byte
+		ValidatorManagerBlockchainID [32]byte
+		ValidatorManagerAddress      common.Address
+		InitialValidators            []InitialValidator
+	}
+	validators := []InitialValidator{}
+	for _, convertSubnetValidator := range convertSubnetValidators {
+		validators = append(validators, InitialValidator{
+			NodeID:       convertSubnetValidator.NodeID[:],
+			BlsPublicKey: convertSubnetValidator.Signer.PublicKey[:],
+			Weight:       convertSubnetValidator.Weight,
+		})
+	}
+	subnetConversionData := SubnetConversionData{
+		SubnetID:                     subnetID,
+		ValidatorManagerBlockchainID: managerBlockchainID,
+		ValidatorManagerAddress:      managerAddress,
+		InitialValidators:            validators,
+	}
+	return contract.TxToMethodWithWarpMessage(
+		rpcURL,
+		privateKey,
+		managerAddress,
+		subnetConversionSignedMessage,
+		"initializeValidatorSet((bytes32,bytes32,address,[(bytes,bytes,uint64)]),uint32)",
+		subnetConversionData,
+		uint32(0),
+	)
+}
+
+// setups PoA manager after a successful execution of
+// ConvertSubnetTx on P-Chain
+// needs the list of validators for that tx,
+// [convertSubnetValidators], together with an evm [ownerAddress]
+// to set as the owner of the PoA manager
+func SetupPoA(
+	app *application.Avalanche,
+	network models.Network,
+	chainSpec contract.ChainSpec,
+	privateKey string,
+	ownerAddress common.Address,
+	convertSubnetValidators []txs.ConvertSubnetValidator,
+) error {
+	rpcURL, _, err := contract.GetBlockchainEndpoints(
+		app,
+		network,
+		chainSpec,
+		true,
+		false,
+	)
+	if err != nil {
+		return err
+	}
+	subnetID, err := contract.GetSubnetID(
+		app,
+		network,
+		chainSpec,
+	)
+	if err != nil {
+		return err
+	}
+	blockchainID, err := contract.GetBlockchainID(
+		app,
+		network,
+		chainSpec,
+	)
+	if err != nil {
+		return err
+	}
+	managerAddress := common.HexToAddress(ValidatorContractAddress)
+	tx, _, err := PoAValidatorManagerInitialize(
+		rpcURL,
+		managerAddress,
+		privateKey,
+		subnetID,
+		ownerAddress,
+	)
+	if err != nil {
+		txHashMsg := ""
+		if tx != nil {
+			txHashMsg = fmt.Sprintf(" (txHash=%s)", tx.Hash().String())
+		}
+		return fmt.Errorf("failure initializing poa validator manager: %w%s", err, txHashMsg)
+	}
+	subnetConversionSignedMessage, err := PoaValidatorManagerGetPChainSubnetConversionWarpMessage(
+		network,
+		app.Log,
+		logging.Info,
+		0,
+		subnetID,
+		blockchainID,
+		managerAddress,
+		convertSubnetValidators,
+	)
+	if err != nil {
+		return fmt.Errorf("failure signing subnet conversion warp message: %w", err)
+	}
+	tx, _, err = PoAValidatorManagerInitializeValidatorsSet(
+		rpcURL,
+		managerAddress,
+		privateKey,
+		subnetID,
+		blockchainID,
+		convertSubnetValidators,
+		subnetConversionSignedMessage,
 	)
-	return err
+	if err != nil {
+		txHashMsg := ""
```

Comment:
nit:  create func for this and move to utils. something like AddHashToErr(tx), so it can be used like
`fmt.Errorf("failure initializing poa validator manager: %w%s", err, utils.AddHashToErr(tx))`